File: Valid Parentheses.js
var isValid = function(s) {
    let stack = []

    for (i=0; i < s.length; i++){

        let bracket= s[i];

        switch(bracket) {
            case '(': stack.push(')');
                break;
            case '[': stack.push(']');
                break;
            case '{': stack.push('}')
                break;
            default:
                
                if (bracket !== stack.pop()) return false;       
        }
    }
    return stack.length === 0;
};

File: 20-valid-parentheses/valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        stack = []
        for i in range(len(s)):
            bracket = s[i]
            
            if bracket == '(':
                stack.append(')')
            elif bracket == '[':
                stack.append(']')
            elif bracket == '{':
                stack.append('}')
            else:
                if not stack or stack.pop() != bracket:
                    return False

        return not stack
             
                
        
        
        